src/QMCHamiltonians/ForwardWalking.h

```diff
@@ -18,99 +18,57 @@
 #ifndef QMCPLUSPLUS_FORWARDWALKING_H
 #define QMCPLUSPLUS_FORWARDWALKING_H
 #include "QMCHamiltonians/OperatorBase.h"
+
 namespace qmcplusplus
 {
 class QMCHamiltonian;
 
-struct ForwardWalking : public OperatorBase
+class ForwardWalking : public OperatorBase
 {
-  std::vector<int> Hindices;
-  std::vector<int> Pindices;
-  std::vector<std::vector<int>> walkerLengths;
-  std::vector<RealType> Values;
-  std::vector<std::string> Names;
-  int blockT, nObservables, nValues, FirstHamiltonian;
-  double count;
-
+public:
   /** constructor
    */
-  ForwardWalking() { update_mode_.set(OPTIMIZABLE, 1); }
+  ForwardWalking();
 
   ///destructor
-  ~ForwardWalking() override {}
+  ~ForwardWalking() override = default;
```

Comment:
Actually I found it beneficial to have the desctructor defined in cpp.
Leave
```
~ForwardWalking() override;
```
in the header file
and put
```
~ForwardWalking() override = default;
```
in the cpp.
When unique_ptr is used, this separation enables forward declaration.